data-structures/ruby/queue.rb

```diff
@@ -0,0 +1,35 @@
+require_relative 'doubly_linked_list/base'
+
+class Queue
+
+  class UnderflowError < StandardError
+  end
+
+  def initialize()
+    @list = DoublyLinkedList::Base.new()
```

Comment:
We don't actually need a doubly linked list to implement a queue.  Remember, although we talk about queues as if there's a "front" or an "end", really what we care about is the fact that we can put information into the queue and later retrieve the oldest information in the queue.

If we're using some kind of list to power these operations, all that means is we need to be able to add elements to one end in constant time and remove them from the other in constant time.  It doesn't matter which end of the list, i.e., there's no reason for us to require that the "front" of the queue corresponds to the "front" of the list.

Singly linked lists always have constant time addition/removal from the head.  We can augment it with a constant time append operation by keeping track of the tail.  Enqueuing would then map to appending information to the end of the list and dequeuing would map to removing information from the head of the list.